controllers/apim/ratelimitpolicy_limits.go

```diff
@@ -115,23 +118,35 @@ func (r *RateLimitPolicyReconciler) gatewayLimits(ctx context.Context, rlpRefs [
 		if rlp.IsForHTTPRoute() {
 			routeRLPList = append(routeRLPList, rlp)
 		} else if rlp.IsForGateway() {
-			gwRLPList = append(gwRLPList, rlp)
+			if gwRLP == nil {
+				gwRLP = rlp
```

Comment:
Is there any difference in the spec between GW RLP and Route RLP?

Reply:
No, the spec is the same. It is only the target. Some RLP target a HTTPRoute and others a Gateway.

The list of RLP's come from some annotations, so here the code just separate them looking at the target.